proxy/envoy/ingress_test.go

```diff
@@ -71,49 +66,9 @@ func compareFile(filename string, golden []byte, t *testing.T) {
 	}
 }
 
-func TestIngressRoutes(t *testing.T) {
-	r := mock.MakeRegistry()
-	s := &mock.SecretRegistry{}
-	mesh := makeMeshConfig()
-	addIngressRoutes(r, t)
-	testIngressConfig(&IngressConfig{
-		Registry:  r,
-		Namespace: ingressNamespace,
-		Secrets:   s,
-		Mesh:      &mesh,
-	}, ingressEnvoyConfig, t)
-}
-
 func TestIngressRoutesSSL(t *testing.T) {
-	r := mock.MakeRegistry()
-	s := &mock.SecretRegistry{"*": ingressTLSSecret}
-	mesh := makeMeshConfig()
-	addIngressRoutes(r, t)
-	testIngressConfig(&IngressConfig{
-		CertFile:  ingressCertFile,
-		KeyFile:   ingressKeyFile,
-		Namespace: ingressNamespace,
-		Secrets:   s,
-		Registry:  r,
-		Mesh:      &mesh,
-	}, ingressEnvoySSLConfig, t)
-	compareFile(ingressCertFile, ingressCert, t)
-	compareFile(ingressKeyFile, ingressKey, t)
-}
-
-func TestIngressRoutesPartialSSL(t *testing.T) {
-	r := mock.MakeRegistry()
-	s := &mock.SecretRegistry{fmt.Sprintf("world.%v.svc.cluster.local", ingressNamespace): ingressTLSSecret}
 	mesh := makeMeshConfig()
-	addIngressRoutes(r, t)
-	testIngressConfig(&IngressConfig{
-		CertFile:  ingressCertFile,
-		KeyFile:   ingressKeyFile,
-		Namespace: ingressNamespace,
-		Secrets:   s,
-		Registry:  r,
-		Mesh:      &mesh,
-	}, ingressEnvoyPartialSSLConfig, t)
+	testIngressOptions(&mesh, ingressTLSSecret, ingressEnvoyConfig, t)
```

Comment:
It looks like `testIngressOptions` is only called once in this function? Were you planning to reuse it for additional test functions? Otherwise it might be good to merge it into TestIngressRouteSSL.